bumble/profiles/vocs.py

```diff
@@ -0,0 +1,244 @@
+
+    @property
+    def audio_location_bytes(self) -> bytes:
+        return struct.pack('<I', self.audio_channel_location)
+
+    @audio_location_bytes.setter
+    def audio_location_bytes(self, new_value: bytes) -> None:
+        self.audio_channel_location = struct.unpack('<I', new_value)[0]
+
+    def _on_read_audio_location(
+        self, _connection: Optional[device.Connection]
+    ) -> bytes:
+        return self.audio_location_bytes
+
+    async def _on_write_audio_location(
+        self, connection: Optional[device.Connection], value: bytes
+    ) -> None:
+        assert connection
+
+        self.audio_location_bytes = value
+        await connection.device.notify_subscribers(
+            attribute=self.audio_channel_location,
+            value=self.audio_location_bytes,
+        )
+        self.emit('audio_location_change')
+
+    async def _on_write_volume_offset_control_point(
+        self, connection: Optional[device.Connection], value: bytes
+    ) -> None:
+        assert connection
+
+        opcode = value[0]
+
+        if opcode != SET_VOLUME_OFFSET_OPCODE:
+            raise att.ATT_Error(ErrorCode.OPCODE_NOT_SUPPORTED)
+
+        change_counter, volume_offset = struct.unpack('<Bh', value[1:])
+
+        if change_counter != self.change_counter:
+            raise att.ATT_Error(ErrorCode.INVALID_CHANGE_COUNTER)
+
+        if not MIN_VOLUME_OFFSET <= volume_offset <= MAX_VOLUME_OFFSET:
+            raise att.ATT_Error(ErrorCode.VALUE_OUT_OF_RANGE)
+
+        self.volume_offset, self.change_counter = (
+            volume_offset,
+            (self.change_counter + 1) % (CHANGE_COUNTER_MAX_VALUE + 1),
+        )
+        await connection.device.notify_subscribers(
+            attribute=self.volume_offset_state,
+            value=self.volume_offset_state_bytes,
+        )
+        self.emit('volume_offset_state_change')
+
+    def _on_read_audio_output_description(self, _connection) -> bytes:
+        return self.audio_output_description_value.encode('utf-8')
+
+    async def _on_write_audio_output_description(self, connection, value) -> None:
+        self.audio_output_description_value = value.decode('utf-8')
+        await connection.device.notify_subscribers(
+            attribute=self.audio_output_description,
+            value=self.audio_output_description_value.encode('utf-8'),
+        )
+        self.emit('audio_output_description_change')
+
+
+# -----------------------------------------------------------------------------
+# Client
+# -----------------------------------------------------------------------------
+class VolumeOffsetControlServiceProxy(gatt_client.ProfileServiceProxy):
+    SERVICE_CLASS = VolumeOffsetControlService
+
+    volume_offset_control_point: gatt_client.CharacteristicProxy
+
+    def __init__(self, service_proxy: gatt_client.ServiceProxy) -> None:
+        self.service_proxy = service_proxy
+
+        self.volume_offset_state = gatt.PackedCharacteristicAdapter(
```

Comment:
It would be nice to create a `VolumeOffsetState` `dataclass` with a `volume_offset` and a `change_counter` property, this would be more user-friendly than an `(int, int)` tuple.